File: leetcode-algorithms/101. Symmetric Tree/SymmetricTree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public bool IsSymmetric(TreeNode root) {
        return IsMirror(root,root);
    }
    public bool IsMirror(TreeNode n1,TreeNode n2){
    if(n1==null&&n2==null)
        return true;
    if(n1==null||n2==null)
        return false;
    return (n1.val==n2.val)&& IsMirror(n1.right,n2.left)&&IsMirror(n1.left,n2.right);

    }
}   

File: leetcode-algorithms/101. Symmetric Tree/SymmetricTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        return self.isMirrorTree(root,root)
    def isMirrorTree (self, t1:TreeNode,t2:TreeNode):
        if(t1 == None and t2 == None):
            return True
        elif(t1 == None or t2 == None):
            return False
        else:
            return (t1.val == t2.val) and self.isMirrorTree(t1.left,t2.right) and self.isMirrorTree(t1.right,t2.left)